rollbar-api/src/main/java/com/rollbar/api/payload/Payload.java

```diff
@@ -0,0 +1,142 @@
+package com.rollbar.api.payload;
+
+import com.rollbar.api.json.JsonSerializable;
+import com.rollbar.api.payload.data.Data;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents the payload to send to Rollbar. A successfully constructed Payload matches Rollbar's
+ * spec, and should be successful when serialized and POSTed to the correct endpoint.
+ */
+public class Payload implements JsonSerializable {
+
+  private final String accessToken;
+
+  private final Data data;
+
+  private Payload(Builder builder) {
+    this.accessToken = builder.accessToken;
+    this.data = builder.data;
+  }
+
+  /**
+   * Getter.
+   * @return the access token.
+   */
+  public String getAccessToken() {
+    return accessToken;
+  }
+
+  /**
+   * Getter.
+   * @return the data.
+   */
+  public Data getData() {
+    return data;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    Payload data = (Payload) o;
+
+    if (accessToken != null ? !accessToken.equals(data.accessToken) : data.accessToken != null) {
+      return false;
+    }
+    return this.data != null ? this.data.equals(data.data) : data.data == null;
+  }
+
+  @Override
+  public int hashCode() {
```

Comment:
Whenever I see custom `hashCode()` functions I get suspicious. How about we create a temp string `accessToken:data` and then take the `hashcode()` of that string?

Reply:
This is how almost every hashCode implementation I've ever seen does it in Java because the built-in string uses something like this pattern with powers of 31 and the sum of individual parts.

Reply:
All hashCode methods have been autogenerated by the IDE using the "facto-standard" way to do it.

Reply:
👍 Good enough for me